### DISCUSSION TITLE
Advantages and disadvantages of Maui

### DISCUSSION THREAD

**Discussion by yangbocheng** [dotnet/maui#13310] [General]
maui is a nice framework, thanks to the open source people.

After i have used maui to make a todolist app. 
currently, i think maui has some advantages and problems.
**Maybe  some understandings below are wrong**

### Advantages
- Single project
- Automatically resize svg
- Powerful vs integration
- Powerful  shell and route
- Similar to wpf
- IOC/DI
- permissions Api

### problems
- Limitation for customizing the ui. there is no controltemplate for button; can't totally customize an input box...
- no routing events：like previewMousedown
- no visual HitTest(?)
- UI is not stable. Fonts and margins are inconsistent on different devices when set a same style

it's too difficult for just want to define a control like 'html div within lots of elements with click event ', then add pressed and unpressed animation is hard.
there are much limitations to make an app which has many changes today

**Comment by davepruitt** [comment]
I would say the main advantages for me are:
1. I get to use C#, which I think is a great language. I'd prefer not to use Javascript in my projects.
2. I enjoy designing my UIs with XAML, because it creates a nice separation of UI and code.

Unfortunately, I do think MAUI is a bit half-baked at the moment. There are currently some big disadvantages (some of them you have pointed out already). For me they are:
1. No XAML previewer, which I think is a very important feature for design-time UI creation
2. Limitations in customizing the UI (as you pointed out). More controls need to have the ability to use ControlTemplate. I want to be able to completely customize a button, and it's annoying that I cannot.
3. While there is a TapGestureRecognizer, it only recognizes taps that have both a "press" and a "release". Sometimes I want to do things "on press" rather than "on tap". There needs to be some sort of "PressGestureRecognizer".
4. Speed - I am developing some pretty basic apps, but they seem to run pretty slowly on my Android devices (I have tested on a Samsung Galaxy Tab S4 and a Samsung Galaxy Tab A7 Lite).
